### PR TITLE
docs(deferring): add type annotation to code examples

### PR DESCRIPTION
**Pull request by Lee-W**
As discussed in https://github.com/apache/airflow/pull/32355#discussion_r1254069649, I'm thinking of adding some type annotations to the documentation.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by pankajkoti — APPROVED** [review]

File: docs/apache-airflow/authoring-and-scheduling/deferring.rst [hunk 1]
```diff
@@ -109,16 +115,22 @@ You are free to set ``method_name`` to ``execute`` if you want your Operator to
 Here's a basic example of how a sensor might trigger deferral::
```
Comment (uranusjr) [inline docs/apache-airflow/authoring-and-scheduling/deferring.rst]: Let’s also use this chance to make these `.. code-block:: python` instead so these are better syntax-highlighted.

**Review by pankajastro — COMMENTED** [review]
we should add the same for example`DateTimeTrigger` too we have 2 serialize and run methods there wdyt?

File: docs/apache-airflow/authoring-and-scheduling/deferring.rst (hunk 1 above)
Comment (Lee-W) [reply docs/apache-airflow/authoring-and-scheduling/deferring.rst]: Sure. Just updated.

**Comment by Lee-W** [comment]
> we should add the same for example`DateTimeTrigger` too we have 2 serialize and run methods there wdyt?

Sorry, I kinda don't get it. Could you point out where is it?

**Comment by pankajastro** [comment]
> Sorry, I kinda don't get it. Could you point out where is it?

sure, type annotation for trigger examples also code block https://github.com/apache/airflow/blob/main/docs/apache-airflow/authoring-and-scheduling/deferring.rst?plain=1#L111-L124

**Comment by Lee-W** [comment]
ah yes, I've changed all the code block into that

File: docs/apache-airflow/authoring-and-scheduling/deferring.rst [hunk 2]
```diff
@@ -56,38 +56,44 @@ Writing a deferrable operator takes a bit more work. There are some main points
...
-* If you want add an operator or sensor that supports both deferrable and non-deferrable modes. It's suggested to add ``deferable: bool = conf.getboolean("operators", "default_deferrable", fallback=False)`` to the ``__init__`` method of the operator and use it to decide whether to run the operator in deferrable mode. You'll be able to configure the default value of ``deferrable`` of all the operators and sensors that supports switch between deferrable and non-deferrable mode through ``default_deferrable`` in the ``operator`` section. Here's an example of a sensor that supports both modes.::
+* If you want add an operator or sensor that supports both deferrable and non-deferrable modes. It's suggested to add ``deferable: bool = conf.getboolean("operators", "default_deferrable", fallback=False)`` to the ``__init__`` method of the operator and use it to decide whether to run the operator in deferrable mode. You'll be able to configure the default value of ``deferrable`` of all the operators and sensors that supports switch between deferrable and non-deferrable mode through ``default_deferrable`` in the ``operator`` section. Here's an example of a sensor that supports both modes.
```
Comment (phanikumv) [inline docs/apache-airflow/authoring-and-scheduling/deferring.rst]:
```suggestion
* If you want add an operator or sensor that supports both deferrable and non-deferrable modes. It's suggested to add ``deferrable: bool = conf.getboolean("operators", "default_deferrable", fallback=False)`` to the ``__init__`` method of the operator and use it to decide whether to run the operator in deferrable mode. You'll be able to configure the default value of ``deferrable`` of all the operators and sensors that supports switch between deferrable and non-deferrable mode through ``default_deferrable`` in the ``operator`` section. Here's an example of a sensor that supports both modes.
```

File: docs/apache-airflow/authoring-and-scheduling/deferring.rst [hunk 3]
```diff
@@ -56,38 +56,44 @@ Writing a deferrable operator takes a bit more work. There are some main points
...
+    from airflow.utils.context import Context
 
 
     class WaitOneHourSensor(BaseSensorOperator):
         def __init__(
-            self,
-            deferable: bool = conf.getboolean("operators", "default_deferrable", fallback=False),
-            **kwargs
-        ):
+            self, deferable: bool = conf.getboolean("operators", "default_deferrable", fallback=False), **kwargs
```
Comment (phanikumv) [inline docs/apache-airflow/authoring-and-scheduling/deferring.rst]:
```suggestion
            self, deferrable: bool = conf.getboolean("operators", "default_deferrable", fallback=False), **kwargs
```

File: docs/apache-airflow/authoring-and-scheduling/deferring.rst [hunk 4]
```diff
@@ -56,38 +56,44 @@ Writing a deferrable operator takes a bit more work. There are some main points
...
-* If you want add an operator or sensor that supports both deferrable and non-deferrable modes. It's suggested to add ``deferable: bool = conf.getboolean("operators", "default_deferrable", fallback=False)`` to the ``__init__`` method of the operator and use it to decide whether to run the operator in deferrable mode. You'll be able to configure the default value of ``deferrable`` of all the operators and sensors that supports switch between deferrable and non-deferrable mode through ``default_deferrable`` in the ``operator`` section. Here's an example of a sensor that supports both modes.::
+* If you want add an operator or sensor that supports both deferrable and non-deferrable modes. It's suggested to add ``deferrable: bool = conf.getboolean("operators", "default_deferrable", fallback=False)`` to the ``__init__`` method of the operator and use it to decide whether to run the operator in deferrable mode. You'll be able to configure the default value of ``deferrable`` of all the operators and sensors that supports switch between deferrable and non-deferrable mode through ``default_deferrable`` in the ``operator`` section. Here's an example of a sensor that supports both modes.
```
Comment (phanikumv) [inline docs/apache-airflow/authoring-and-scheduling/deferring.rst]: Here we are talking about both operators and sensors, but giving an example of `deferrable: bool = conf.getboolean("operators", "default_deferrable", fallback=False)` only for operators.

Comment (syedahsn) [inline docs/apache-airflow/authoring-and-scheduling/deferring.rst]: small nitpick
```suggestion
* If you want to add an operator or sensor that supports both deferrable and non-deferrable modes, it's suggested to add ``deferrable: bool = conf.getboolean("operators", "default_deferrable", fallback=False)`` to the ``__init__`` method of the operator and use it to decide whether to run the operator in deferrable mode. You'll be able to configure the default value of ``deferrable`` of all the operators and sensors that support switching between deferrable and non-deferrable mode through ``default_deferrable`` in the ``operator`` section. Here's an example of a sensor that supports both modes.
```

**Review by syedahsn — APPROVED** [review]
LGTM

File: docs/apache-airflow/authoring-and-scheduling/deferring.rst (hunk 4 above)
Comment (Lee-W) [reply docs/apache-airflow/authoring-and-scheduling/deferring.rst]: As the sensor is a kind of operator as well https://github.com/apache/airflow/blob/main/airflow/sensors/base.py#L82, it seems to me this should be fine. But I'm good with making this change as well. We probably should add `default_deferrable` [here](https://github.com/apache/airflow/blob/main/airflow/config_templates/config.yml#L2943). Maybe that should be another PR?

**Review by potiuk — APPROVED** [review]

**Pull request merged**